agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/AgendaDetailScreen.kt

```diff
@@ -71,6 +71,9 @@ fun AgendaDetailScreenRoot(
             viewModel.onAction(AgendaDetailAction.OnDisableEdit)
         }
     }
+    LaunchedEffect(type) {
+        viewModel.onAction(AgendaDetailAction.OnSetAgendaItemType(type))
+    }
```

Comment:
Similar as my answer to your other comment (https://github.com/evanadwyer-pronoid-software/Tasky/pull/81#discussion_r1701763767), you could retrieve the type in the ViewModel directly by using SavedStateHandle

Reply:
ah, forgot that hilt will automatically wire this together to make it accessible, that's way better, thanks!